tasks/alignment/task_bwa.wdl

```diff
@@ -25,25 +27,98 @@ task bwa {
       ref_genome="/artic-ncov2019/primer_schemes/nCoV-2019/V3/nCoV-2019.reference.fasta"  
     fi
 
-    # Map with BWA MEM
-    echo "Running bwa mem -t ~{cpu} ${ref_genome} ~{read1} ~{read2} | samtools sort | samtools view -F 4 -o ~{samplename}.sorted.bam "
+    # set cat command based on compression
+    if [[ "~{read1}" == *".gz" ]] ; then
+      cat_reads="zcat"
+    else
+      cat_reads="cat"
+    fi
+
+    echo -e "\ninput R1 has $(${cat_reads} ~{read1} | grep -c '^@') reads as input"
+    echo "input R2 has $(${cat_reads} ~{read2} | grep -c '^@') reads as input"
+
+    # Map with BWA MEM; pipe to samtools sort to write sorted SAM file
     bwa mem \
-    -t ~{cpu} \
-    "${ref_genome}" \
-    ~{read1} ~{read2} |\
-    samtools sort | samtools view -F 4 -o ~{samplename}.sorted.bam
+      -t ~{cpu} \
+      "${ref_genome}" \
+      ~{read1} \
+      ~{read2} | \
+    samtools sort \
+      -@ ~{cpu} - \
+      > ~{samplename}.sorted.sam
+    
+    # convert SAM to BAM that only includes aligned reads
+    samtools view \
+      -@ ~{cpu} \
+      -F 4 \
+      -b \
+      -o ~{samplename}.sorted.bam \
```

Comment:
shouldn't this file be renamed to `~{samplename}.sorted.aligned-reads.bam`?

Reply:
I kept it as the original filename. I agree it would provide a bit more clarity to someone reading the code, but hopefully the comments are enough. 

I would prefer not to make further changes unless truly necessary since we'd like to share with our partner asap